[BUG] `test_cast_timestamp_to_date` fails with `TZ=Asia/Hebron`

**Describe the bug**

```
FAILED ../../src/main/python/cast_test.py::test_cast_timestamp_to_date[DATAGEN_SEED_OVERRIDE=1704816489] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
```

**Steps/Code to reproduce bug**

```
export TZ=Asia/Hebron
./integration_tests/run_pyspark_from_build.sh -k test_cast_timestamp_to_date
```

**Expected behavior**


**Environment details (please complete the following information)**
CI and local workstation. Spark 3.1.1

**Additional context**

--- a/integration_tests/src/main/python/cast_test.py
+++ b/integration_tests/src/main/python/cast_test.py
@@ -15,7 +15,7 @@
 import pytest
 
 from asserts import *
-from conftest import is_not_utc
+from conftest import is_not_utc, is_supported_time_zone
 from data_gen import *
 from spark_session import *
 from marks import allow_non_gpu, approximate_float, datagen_overrides, tz_sensitive_test
@@ -540,6 +540,8 @@ def test_cast_timestamp_to_string():
         lambda spark: unary_op_df(spark, timestamp_gen)
             .selectExpr("cast(a as string)"))
 
+@tz_sensitive_test
+@allow_non_gpu(*non_supported_tz_allow)
 def test_cast_timestamp_to_date():
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: unary_op_df(spark, timestamp_gen)

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -24,7 +24,7 @@ import random
 from spark_session import is_before_spark_340, with_cpu_session
 import sre_yield
 import struct
-from conftest import skip_unless_precommit_tests, get_datagen_seed, is_not_utc
+from conftest import skip_unless_precommit_tests, get_datagen_seed, is_not_utc, is_supported_time_zone
 import time
 import os
 from functools import lru_cache
@@ -1211,6 +1211,11 @@ def get_25_partitions_df(spark):
 non_utc_allow = ['ProjectExec', 'FilterExec', 'FileSourceScanExec', 'BatchScanExec', 'CollectLimitExec',
                  'DeserializeToObjectExec', 'DataWritingCommandExec', 'WriteFilesExec', 'ShuffleExchangeExec',
                  'ExecutedCommandExec'] if is_not_utc() else []
+
+non_supported_tz_allow = ['ProjectExec', 'FilterExec', 'FileSourceScanExec', 'BatchScanExec', 'CollectLimitExec',
+                 'DeserializeToObjectExec', 'DataWritingCommandExec', 'WriteFilesExec', 'ShuffleExchangeExec',
+                 'ExecutedCommandExec'] if not is_supported_time_zone() else []
+
 
 # date related regexps for generating date strings within python's range limits
 

